cmd/yorkie/version.go

```diff
@@ -17,27 +17,162 @@
+		PreRunE: config.Preload,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			fmt.Printf("Yorkie: %s\n", version.Version)
-			fmt.Printf("Go: %s\n", runtime.Version())
-			fmt.Printf("Build date: %s\n", version.BuildDate)
+			if err := Validate(); err != nil {
+				return err
+			}
+
+			var versionInfo types.VersionInfo
+			versionInfo.ClientVersion = getYorkieClientVersion()
+
+			serverVersionChan := make(chan *types.VersionDetail)
+			errorChan := make(chan error)
+
+			if !clientOnly {
+				go func() {
+					rpcAddr := viper.GetString("rpcAddr")
+					auth, err := config.LoadAuth(rpcAddr)
+					if err != nil {
+						errorChan <- err
+						return
+					}
+
+					cli, err := admin.Dial(rpcAddr, admin.WithToken(auth.Token), admin.WithInsecure(auth.Insecure))
+					if err != nil {
+						errorChan <- err
+						return
+					}
+					defer cli.Close()
+
+					ctx := context.Background()
+					sv, err := cli.GetServerVersion(ctx)
+					if err != nil {
+						errorChan <- err
+						return
+					}
+
+					serverVersionChan <- sv
+				}()
+			}
+
+			var serverErr error
+
+			if !clientOnly {
+				select {
+				case sv := <-serverVersionChan:
+					versionInfo.ServerVersion = sv
+				case err := <-errorChan:
+					serverErr = err
+				}
+			}
+
+			switch output {
+			case "":
+				cmd.Printf("Yorkie Client: %s\n", versionInfo.ClientVersion.YorkieVersion)
+				cmd.Printf("Go: %s\n", versionInfo.ClientVersion.GoVersion)
+				cmd.Printf("Build Date: %s\n", versionInfo.ClientVersion.BuildDate)
+				if versionInfo.ServerVersion != nil {
+					cmd.Printf("Yorkie Server: %s\n", versionInfo.ServerVersion.YorkieVersion)
+					cmd.Printf("Go: %s\n", versionInfo.ServerVersion.GoVersion)
+					cmd.Printf("Build Date: %s\n", versionInfo.ServerVersion.BuildDate)
+				}
+			case "yaml":
+				marshalled, err := yaml.Marshal(&versionInfo)
+				if err != nil {
+					return errors.New("failed to marshal YAML")
+				}
+				fmt.Println(string(marshalled))
+			case "json":
+				marshalled, err := json.MarshalIndent(&versionInfo, "", "  ")
+				if err != nil {
+					return errors.New("failed to marshal JSON")
+				}
+				fmt.Println(string(marshalled))
+			}
+
+			if serverErr != nil {
+				cmd.Printf("Error fetching server version: ")
+
+				connectErr := new(connect.Error)
+				if errors.As(serverErr, &connectErr) {
+					// TODO(hyun98): Find cases where different error cases can occur,
+					// and display a user-friendly error message for each case.
+					// Furthermore, it would be good to improve it
+					// by creating a general-purpose error handling module for rpc communication.
+					// for example, rpc error handling reference: https://connectrpc.com/docs/go/errors/
+					switch connectErr.Code() {
+					case connect.CodeUnimplemented:
+						cmd.Printf("The server does not support this operation. You might need to check your server version.\n")
+					}
+				} else {
+					cmd.Print(serverErr)
+				}
+			}
+
```

Comment:
I referred to the connect-rpc documentation as suggested by @krapie and displayed a user-friendly message when the yorkie server could not response the updated version command.

In my opinion, if this pr is released and someone's yorkie client version is updated, the user will not know the cause if the version command only shows `unimplemented: 404 Not Found`(ambiguous default error message).
Therefore, only `unimplemented` errors that were likely to have a significant impact on users were handled as separate messages.

Finally, I left an additional TODO comment, and it would be good to work on improving the remaining error cases and rpc exception handling module in another issue.